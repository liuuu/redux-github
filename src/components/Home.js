// @flow
import React from "react";
import QueueAnim from "rc-queue-anim";
import { Link } from "react-router-dom";
import facebook from "../img/facebook-icon.svg";
import github from "../img/github-icon-1.svg";
import react1 from "../img/react-1.svg";

class Home extends React.Component {
  state: {
    show: boolean
  };

  constructor() {
    super();
    this.state = {
      show: false
    };
  }

  componentDidMount() {
    this.setState({
      show: true
    });
  }

  render() {
    return (
      <div className="home">
        <div className="home-header" key="header">
          <h1>This is a redux Demo site</h1>
        </div>

        <QueueAnim
          className="democontent"
          duration={1500}
          delay={100}
          type="bottom"
        >
          {this.state.show
            ? [
                <ul key="box" className="box">
                  <li key="0">
                    <img src={facebook} alt="sdf" />
                  </li>
                  <li key="1">
                    <img src={react1} alt="github-icon.svg" />
                  </li>
                  <li key="2">
                    <img src={github} alt="react-1.svg" />
                  </li>
                </ul>
              ]
            : null}
        </QueueAnim>
        <div key="button">
          <button>
            <Link to="all">browser the popular Projects of github</Link>
          </button>
        </div>
      </div>
    );
  }
}

export default Home;
