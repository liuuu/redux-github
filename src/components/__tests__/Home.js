import React from "react";
import renderer from "react-test-renderer";
import Home from "../Home";
import toJson from "enzyme-to-json";
import { render, shallow, mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";

it("Home renders correctly", () => {
  const wrapper = render(<MemoryRouter><Home /></MemoryRouter>);
  expect(toJson(wrapper)).toMatchSnapshot();
});
