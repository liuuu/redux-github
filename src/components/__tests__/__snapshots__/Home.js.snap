// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  class="home"
>
  <div
    class="home-header"
  >
    <h1>
      This is a redux Demo site
    </h1>
  </div>
  <div
    class="democontent"
  />
  <div>
    <button>
      <a
        href="all"
      >
        browser the popular Projects of github
      </a>
    </button>
  </div>
</div>
`;
